src/composed/message/types.rs

```diff
@@ -433,34 +438,57 @@ impl Message {
                     SignatureType::Text
                 };
 
-                let signature_config = SignatureConfig::new_v4(
-                    Default::default(),
-                    typ,
-                    algorithm,
-                    hash_algorithm,
-                    hashed_subpackets,
-                    unhashed_subpackets,
-                );
-                (typ, signature_config.sign(key, key_pw, l.data())?)
+                let mut config = match sig_version {
+                    SignatureVersion::V4 => SignatureConfig::v4(typ, algorithm, hash_algorithm),
+                    SignatureVersion::V6 => {
+                        SignatureConfig::v6_with_rng(rng, typ, algorithm, hash_algorithm)?
```

Comment:
not a fan of this name, maybe make `v6` by default take an `rng` and make a `v6_with_salt`?

Reply:
yes, sounds good (71070ad0ff4fcaba297cc697fc6347cb30827e31)